### PR TITLE
Maykumar/feature operations

### PR DESCRIPTION
**Pull request by MayankKumar91**
# Description

1. Added schema for Featurestore, Featureset and Featurestore entity
2. Added load method for featurestore, featureset and featurestore_entity
3. Added upload artifact functionality for featureset
4. Added unit test for load methods

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/feature_schema.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from azure.ai.ml._schema.core.schema import PatchedSchemaMeta
+
+
+class FeatureSchema(metaclass=PatchedSchemaMeta):
+    name = fields.Str(
+        required=True,
+        allow_none=False,
+    )
+    type = fields.Str(
+        validate=validate.OneOf(["string", "integer", "long", "float", "double", "binary", "datetime", "boolean"]),
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/feature_schema.py:18]: Is it easier to have a enum type here?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/featureset_spec_schema.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from azure.ai.ml._schema.core.fields import NestedField
+from azure.ai.ml._schema.core.schema import YamlFileSchema
+from azure.ai.ml._schema._featurestore_entity.data_column_schema import DataColumnSchema
+
+from .source_metadata_schema import SourceMetadataSchema
+from .delay_metadata_schema import DelayMetadataSchema
+from .feature_schema import FeatureSchema
+
+
+class FeaturesetSpecSchema(YamlFileSchema):
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/featureset_spec_schema.py:20]: missing `TransformationCode`?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/timestamp_column_metadata_schema.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# ---------------------------------------------------------
+
+# pylint: disable=unused-argument,no-self-use
+
+from marshmallow import fields, post_load
+
+from azure.ai.ml._schema.core.schema import PatchedSchemaMeta
+
+
+class TimestampColumnMetadataSchema(metaclass=PatchedSchemaMeta):
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/timestamp_column_metadata_schema.py:12]: name is required, format is optional

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/feature_schema.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        required=True,
+        allow_none=False,
+    )
+    description = fields.Str(required=False)
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/feature_schema.py:22]: missing `tags`?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/feature_schema.py (hunk 1 above)
Comment (MayankKumar91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_featureset/feature_schema.py:18]: in sdk, even though we have enum type, we allow string to be as input

File: sdk/ml/azure-ai-ml/azure/ai/ml/__init__.py [hunk 5]
```diff
@@ -57,6 +60,9 @@
     "load_datastore",
     "load_model",
     "load_environment",
+    "load_featureset",
+    "load_feature_store",
```
Comment (singankit) [inline sdk/ml/azure-ai-ml/azure/ai/ml/__init__.py:64]: load_feature_store vs load_featurestore
Would be good to make this and load_featurestore_entity consistent

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_featureset/featureset_spec.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY
+from azure.ai.ml.entities._util import load_from_dict
+from azure.ai.ml.entities._featurestore_entity.data_column import DataColumn
+
+from .feature import Feature
+from .source_metadata import SourceMetadata
+from .delay_metadata import DelayMetadata
+
+
+class FeaturesetSpec(object):
```
Comment (singankit) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_featureset/featureset_spec.py:23]: Shall we spell out the specification fully ?

Comment (MayankKumar91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_featureset/featureset_spec.py:23]: there is another class called FeaturesetSpecification

File: sdk/ml/azure-ai-ml/azure/ai/ml/__init__.py (hunk 5 above)
Comment (MayankKumar91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/__init__.py:64]: the classes are like :
FeatureStore
Featureset
FeaturestoreEntity

Comment (MayankKumar91) [reply sdk/ml/azure-ai-ml/azure/ai/ml/__init__.py:64]: Could be updated later after discussion with feature store crew

**Review by bastrik — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/__init__.py [hunk 7]
```diff
@@ -24,6 +24,9 @@
     load_data,
     load_datastore,
     load_environment,
+    load_featureset,
+    load_feature_store,
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/__init__.py:28]: Should featurstore be one word or 2 words? Let's be consistent.

**Review by luigiw — APPROVED** [review]

**Pull request merged**